sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpressions.scala

```diff
@@ -145,21 +173,14 @@ trait GpuBinaryExpression extends BinaryExpression with GpuExpression {
         (lhs, rhs) match {
           case (l: GpuColumnVector, r: GpuColumnVector) =>
             GpuColumnVector.from(doColumnar(l, r), dataType)
-          case (l, r: GpuColumnVector) =>
-            withResource(GpuScalar.from(l, left.dataType)) { scalar =>
-              GpuColumnVector.from(doColumnar(scalar, r), dataType)
-            }
-          case (l: GpuColumnVector, r) =>
-            withResource(GpuScalar.from(r, right.dataType)) { scalar =>
-              GpuColumnVector.from(doColumnar(l, scalar), dataType)
-            }
-          case (l, r) if l != null && r != null =>
-            withResource(GpuScalar.from(l, left.dataType)) { leftScalar =>
-              withResource(GpuScalar.from(r, right.dataType)) { rightScalar =>
-                GpuColumnVector.from(doColumnar(batch.numRows(), leftScalar, rightScalar), dataType)
-              }
-            }
-          case _ => null
+          case (l: Scalar, r: GpuColumnVector) =>
+            GpuColumnVector.from(doColumnar(l, r), dataType)
+          case (l: GpuColumnVector, r: Scalar) =>
+            GpuColumnVector.from(doColumnar(l, r), dataType)
+          case (l: Scalar, r: Scalar) =>
+            GpuColumnVector.from(doColumnar(batch.numRows(), l, r), dataType)
+          // null is not welcome, use a null Column.
+          case _ => GpuColumnVector.fromNull(batch.numRows(), dataType)
```

Comment:
IMO it's dangerous to assume anything else that didn't match must be null.  I'd rather see a check for it here just in case some other type slips through.

Reply:
Yes you are right. Added the null check.